### ISSUE TITLE
monodevelop lunching System.TypeInitializationException: error

### ISSUE THREAD

**Issue by maxamin** [mono/monodevelop#7072]
hello guys im new in c# and i installed monodevelop on my parrotsec os and it did not work this is what it prints
parrot]─[~]
└──╼ $monodevelop

```
ERROR [2019-02-06 07:23:21Z]: Could not purge old log files
ERROR [2019-02-06 07:23:21Z]: Could not read branding document
System.TypeInitializationException: The type initializer for 'MonoDevelop.Core.FilePath' threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at System.Collections.Generic.LargeArrayBuilder`1[T].GetBuffer (System.Int32 index) [0x00022] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.LargeArrayBuilder`1[T].CopyTo (System.Collections.Generic.CopyPosition position, T[] array, System.Int32 arrayIndex, System.Int32 count) [0x00041] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].CopyTo (T[] array, System.Int32 arrayIndex, System.Int32 count) [0x0009a] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].ToArray () [0x00028] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Linq.Enumerable+Concat2Iterator`1[TSource].ToArray () [0x00024] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at MonoDevelop.Core.FilePath..cctor () [0x00071] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
   --- End of inner exception stack trace ---
  at MonoDevelop.Core.UserProfile.GetProfile (System.String profileVersion) [0x0000b] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
  at MonoDevelop.Core.UserProfile..cctor () [0x00044] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
ERROR [2019-02-06 07:23:21Z]: Failed to redirect output to log file
System.TypeInitializationException: The type initializer for 'MonoDevelop.Core.UserProfile' threw an exception. ---> System.TypeInitializationException: The type initializer for 'MonoDevelop.Core.FilePath' threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at System.Collections.Generic.LargeArrayBuilder`1[T].GetBuffer (System.Int32 index) [0x00022] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.LargeArrayBuilder`1[T].CopyTo (System.Collections.Generic.CopyPosition position, T[] array, System.Int32 arrayIndex, System.Int32 count) [0x00041] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].CopyTo (T[] array, System.Int32 arrayIndex, System.Int32 count) [0x0009a] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].ToArray () [0x00028] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Linq.Enumerable+Concat2Iterator`1[TSource].ToArray () [0x00024] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at MonoDevelop.Core.FilePath..cctor () [0x00071] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
   --- End of inner exception stack trace ---
  at MonoDevelop.Core.UserProfile.GetProfile (System.String profileVersion) [0x0000b] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
  at MonoDevelop.Core.UserProfile..cctor () [0x00044] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
   --- End of inner exception stack trace ---
  at MonoDevelop.Core.LoggingService.RedirectOutputToFileUnix () [0x00001] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
  at MonoDevelop.Core.LoggingService.RedirectOutputToLogFile () [0x00019] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
FATAL ERROR [2019-02-06 07:23:21Z]: MonoDevelop failed to start. Some of the assemblies required to run MonoDevelop (for example gtk-sharp)may not be properly installed in the GAC.
System.TypeInitializationException: The type initializer for 'MonoDevelop.Core.FilePath' threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at System.Collections.Generic.LargeArrayBuilder`1[T].GetBuffer (System.Int32 index) [0x00022] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.LargeArrayBuilder`1[T].CopyTo (System.Collections.Generic.CopyPosition position, T[] array, System.Int32 arrayIndex, System.Int32 count) [0x00041] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].CopyTo (T[] array, System.Int32 arrayIndex, System.Int32 count) [0x0009a] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Collections.Generic.SparseArrayBuilder`1[T].ToArray () [0x00028] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Linq.Enumerable+Concat2Iterator`1[TSource].ToArray () [0x00024] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00021] in <13c0993ff82548209b09f271bd5e6a78>:0 
  at MonoDevelop.Core.FilePath..cctor () [0x00071] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
   --- End of inner exception stack trace ---
  at MonoDevelop.Core.SystemInformation.GetReleaseId () [0x0000b] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
  at MonoDevelop.Core.Runtime.get_Version () [0x00021] in <a9ddf35690194f9ba9d5cc07a27eaff9>:0 
  at MonoDevelop.Ide.IdeVersionInfo.get_MonoDevelopVersion () [0x00007] in <de8602061f72461ba5874c810930ad23>:0 
  at MonoDevelop.Ide.IdeStartup.Run (MonoDevelop.Ide.MonoDevelopOptions options) [0x00001] in <de8602061f72461ba5874c810930ad23>:0 
  at MonoDevelop.Ide.IdeStartup.Main (System.String[] args, MonoDevelop.Ide.Extensions.IdeCustomizer customizer) [0x00090] in <de8602061f72461ba5874c810930ad23>:0 
```

**Comment by Nable80** [comment]
I have the same problem under Debian Testing.
As far as I understand, ``BrandingService`` constructor is broken: it doesn't handle absence of ``branding`` directory or ``Branding.xml`` file correctly.

Comment suggests that ``ApplicationName`` is never ``null`` but it's not true if one lacks ``../branding/Branding.xml`` file. Thus we have NRE here:
https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Core/MonoDevelop.Core/BrandingService.cs#L89-L123

**Comment by Nable80** [comment]
Dear developers,
would submitting a pull-request increase the chance to get this issue finally fixed?
Should I spend time on preparing the patch or it's unlikely to be accepted?
This problem completely prevents me from using monodevelop for a long-long time. Most probably it's not just me.

**Comment by Therzok** [comment]
@Nable80 I am sorry for missing this issue. I had it opened on a phone and it gto lost.

**Comment by Therzok** [comment]
@Nable80 Can you test this? https://github.com/mono/monodevelop/pull/7589

**Comment by Nable80** [comment]
@Therzok Thank you very much for the quick reply and this fix!
I don't have a working build environment at the moment but I hope I can prepare it ~tomorrow.

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.2.0.311 (master)

Pull Request #7589 merged by: Lluis Sanchez
Author: mono
Commit: 3cdacf922320fa4f299f26a94ec97a4d50b2aeda (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7589 merged by: Lluis Sanchez
Author: mono
Commit: 3cdacf922320fa4f299f26a94ec97a4d50b2aeda (xamarin/monodevelop)
